### DISCUSSION TITLE
Boa javascript engine support

### DISCUSSION THREAD

**Discussion by yavko** [servo/servo#29100] [Ideas] [answered]
Would be cool to see the https://github.com/boa-dev/boa javascript engine be adopted, maybe just as a build flag, or something the like

**Comment by jdm** [comment, marked as answer]
The integration with the JS engine is very deep, and switching engines would require a huge amount of work. It's not going to happen without a really obvious reason why the new engine would be an improvement.

**Reply by Endo-Heikichi** [reply]
> switching engines would require a huge amount of work.

Just to double check. Does Servo currently use a custom JS engine built in Rust or does it have bindings to Mozilla's Spidermonkey?

**Reply by yavko** [reply]
> > switching engines would require a huge amount of work.
> 
> Just to double check. Does Servo currently use a custom JS engine built in Rust or does it have bindings to Mozilla's Spidermonkey?

spidermonkey bindings

**Reply by Endo-Heikichi** [reply]
> spidermonkey bindings

Appreciate the quick response. Instead of leveraging Spidermonkey, imo a pure Rust JS engine would be more interesting. Obviously Boa isn't as battle tested as V8 or Spidermonkey but it is in Rust and is being actively worked upon. 

I understand that developing browser engines these days is just about as complex as designing OSs due to the ever increasing complexity of web standards. I mean despite Rust being such a wonderful language, there have been no attempts at creating browser engines in it from scratch since Servo. That's a testament to just how complex browser development is. So I appreciate the devs continuing to work on it. It was just my desire to see the entire browser stack (browser engine and JS engine) in Rust.
